src/parser/syntax.ml

```diff
@@ -118,13 +118,17 @@ module Table = struct
 end
 
 module Reader = struct
-  let until_rbrace input acc =
+  let until_rbrace_or_eof input acc =
     let rec consume () =
       let next_token = peek input in
       match next_token.value with
       | `Right_brace ->
           junk input;
           `End (acc, next_token.location)
+      | `End ->
+          Parse_error.unclosed_table next_token.location |> add_warning input;
```

Comment:
This function seems to only be used when parsing tables but its name doesn't say that it's specific to tables. Could the error message be changed not to be specific to tables ? Perhaps by taking one more argument here.

Reply:
Indeed, thanks